File: RyuzakiLib/db/pymongo.py
from pymongo import MongoClient


class MongoConnect:
    def __init__(self, mongo_url=None, client=None, database=None):
        self.mongo_url = mongo_url
        self.client = client
        self.database = database

    def get_collection(self, mongodb_dev: bool = None):
        client_mongo = MongoClient(self.mongo_url)
        if mongodb_dev:
            db = client_mongo[self.client][self.database]
        else:
            db = client_mongo["tiktokbot"]["ryuzakilib"]

        return db[self.collection_name]

File: cpp/pymongo.cpp
#include <iostream>
#include <mongocxx/client.hpp>
#include <mongocxx/instance.hpp>
#include <mongocxx/uri.hpp>

class MongoConnect {
public:
    MongoConnect(std::string mongo_url = "", std::string client = "", std::string database = "", bool mongodb_connect = false)
        : mongo_url(mongo_url), client(client), database(database), mongodb_connect(mongodb_connect) {}

    mongocxx::collection get_collection() {
        if (mongodb_connect) {
            mongocxx::uri uri(mongo_url);
            mongocxx::client client(uri);
            mongocxx::database db = client[client];
            mongocxx::collection collection = db[database];
            return collection;
        } else {
            mongocxx::uri uri(mongo_url);
            mongocxx::client client(uri);
            mongocxx::database db = client["tiktokbot"];
            mongocxx::collection collection = db["ryuzakilib"];
            return collection;
        }
    }

private:
    std::string mongo_url;
    std::string client;
    std::string database;
    bool mongodb_connect;
};
